tox.ini

```diff
@@ -107,6 +108,20 @@ commands =
     pre-commit install-hooks
     pre-commit run --verbose --all-files --show-diff-on-failure
 
+[testenv:base_deps]
+description = Check the test suite does not fail if there are not extra dependencies
+basepython = python3.8
+extras =
+deps =
+  numpy>=1.16.0
+  scipy>=1.2.0
+  matplotlib>=2.2.2
+  pandas>=0.23.0
+  astropy>=4.0
+  parfive[ftp]>=1.1.0
+commands =
+    python -c "import sunpy; sunpy.self_test()"
```

Comment:
Are we actually verifying anything with this test though? What exit status do we return if deps are missing?

Reply:
Just wanted to make sure it worked and using tox seemed the easiest thing to do.

Reply:
LIke so;
```
.package create: /home/nabil/GitHub/sunpy/.tox/.package
.package installdeps: setuptools, setuptools_scm, wheel, oldest-supported-numpy, extension-helpers
base_deps create: /home/nabil/GitHub/sunpy/.tox/base_deps
base_deps installdeps: numpy>=1.16.0, scipy>=1.2.0, matplotlib>=2.2.2, pandas>=0.23.0, astropy>=4.0, parfive[ftp]>=1.1.0
base_deps inst: /home/nabil/GitHub/sunpy/.tox/.tmp/package/1/sunpy-2.1.dev782+g9b4b4adc7.d20201121.tar.gz
base_deps installed: aioftp==0.18.1,aiohttp==3.7.3,astropy==4.1,async-timeout==3.0.1,attrs==20.3.0,chardet==3.0.4,cycler==0.10.0,h5netcdf==0.8.1,h5py==3.1.0,idna==2.10,kiwisolver==1.3.1,matplotlib==3.3.3,multidict==5.0.2,numpy==1.19.4,pandas==1.1.4,parfive==1.2.0,Pillow==8.0.1,pyparsing==2.4.7,python-dateutil==2.8.1,pytz==2020.4,scipy==1.5.4,six==1.15.0,sunpy @ file:///home/nabil/GitHub/sunpy/.tox/.tmp/package/1/sunpy-2.1.dev782%2Bg9b4b4adc7.d20201121.tar.gz,tqdm==4.52.0,typing-extensions==3.7.4.3,yarl==1.6.3
base_deps run-test-pre: PYTHONHASHSEED='1235638279'
base_deps run-test: commands[0] | python -c 'import sunpy; sunpy.self_test()'



Starting sunpy self test...
Checking for packages needed to run sunpy:
The following packages are not installed for the sunpy[database] requirement:
  * sqlalchemy
The following packages are not installed for the sunpy[image] requirement:
  * scikit-image
The following packages are not installed for the sunpy[jpeg2000] requirement:
  * glymur
The following packages are not installed for the sunpy[net] requirement:
  * drms
  * beautifulsoup4
  * zeep
You do not have all the required dependencies installed to run the sunpy test suite.
NOTE THIS JUST MEANS THE TESTS WILL NOT RUN, NOT THAT SUNPY MISCONFIGURED.
If you want to run the sunpy tests install the 'tests' extra with `pip install sunpy[all,tests]`
_________________________________________________________________________________________________________________________________________________________ summary __________________________________________________________________________________________________________________________________________________________
  base_deps: commands succeeded
  congratulations :)
```